Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebView.cpp

```diff
@@ -1934,6 +1934,54 @@ static void testWebViewWebExtensionMode(WebViewTest* test, gconstpointer)
     g_assert_true(WebViewTest::javascriptResultToBoolean(javascriptResult));
 }
 
+static void testWebViewDisableWebSecurity(WebViewTest* test, gconstpointer)
+{
+    webkit_web_context_register_uri_scheme(test->m_webContext.get(), "foo",
+        [](WebKitURISchemeRequest* request, gpointer userData) {
+            GRefPtr<GInputStream> inputStream = adoptGRef(g_memory_input_stream_new());
+            const char* data = "<p>foobar!</p>";
+            g_memory_input_stream_add_data(G_MEMORY_INPUT_STREAM(inputStream.get()), data, strlen(data), nullptr);
+            webkit_uri_scheme_request_finish(request, inputStream.get(), strlen(data), "text/html");
+        }, nullptr, nullptr);
+
+    char html[] = "<html><script>let foo = 0; fetch('foo://bar/baz').then(response => { foo = response.status; }).catch(err => { foo = -1; });</script></html>";
+
+    auto waitForFooChanged = [&test]() {
+        GUniqueOutPtr<GError> error;
+        WebKitJavascriptResult* result;
+        JSCValue* jscvalue;
+        int value;
+        do {
+            result = test->runJavaScriptAndWaitUntilFinished("foo;", &error.outPtr());
+            g_assert_no_error(error.get());
+            jscvalue = webkit_javascript_result_get_js_value(result);
+            value = jsc_value_to_int32(jscvalue);
+            webkit_javascript_result_unref(result);
+        } while (!value);
```

Comment:
This is a bit confusing, I guess this is 0 when the JSCValue is undefined. I would explicitly check if value is undefined instead.

Reply:
jscvalue is 0 if the fetch has not finished yet(initial value of 'foo'). If  fetch finished, then 'foo' is set to `response.status` in case of success and to -1 in case any problems.
<script>let foo = 0; fetch('foo://bar/baz').then(response => { foo = response.status; }).catch(err => { foo = -1; });</script>